src/Functions/parseDateTime.cpp

```diff
@@ -684,57 +694,116 @@ namespace
                     arguments[0].column->getName(),
                     getName());
 
-            String format = getFormat(arguments);
-            const auto & time_zone = getTimeZone(arguments);
+            ColumnUInt8::MutablePtr col_null_map;
+            if constexpr (error_handling == ErrorHandling::Null)
+                col_null_map = ColumnUInt8::create(input_rows_count, 0);
+
+            Int64 multiplier = 0;
+            UInt32 scale = 0;
+            if constexpr (return_type == ReturnType::DateTime64)
+            {
+                const DataTypeDateTime64 * datatime64_type = checkAndGetDataType<DataTypeDateTime64>(removeNullable(result_type).get());
+                scale = datatime64_type->getScale();
+                multiplier = DecimalUtils::scaleMultiplier<DateTime64>(scale);
+            }
+
+            const String format = getFormat(arguments, scale);
             std::vector<Instruction> instructions = parseFormat(format);
+            Int64OrError result = 0;
 
+            /// Check scale by counting how many 'S' characters exist in the format string when parse syntax is Joda and
+            /// return type is DateTime64.
+            if constexpr (return_type == ReturnType::DateTime64 && parse_syntax == ParseSyntax::Joda)
+            {
+                UInt32 s_count = 0;
+                for (const auto & instruction : instructions)
+                {
+                    const String & fragment = instruction.getFragment();
+                    for (char ch : fragment)
+                    {
+                        if (ch != 'S')
+                        {
+                            s_count = 0;
+                            break;
+                        }
+                        else
+                            ++s_count;
+                    }
+                }
+                /// If the number of 'S' characters in format string not euqals the scale, then throw an exception to report error.
+                if (s_count != 0 && s_count != scale)
+                    result = tl::unexpected(ErrorCodeAndMessage(ErrorCodes::BAD_ARGUMENTS,
```

Comment:
Here, `result` is set to an error but I don't see where we will throw or abort execution. We will enter the loop (l. 744) and ignore the error.